src/library/entities/Patron.java

```diff
@@ -1,118 +1,116 @@
-                
-    }
-    
-    public String GeT_LaSt_NaMe() {
-        return LaSt_NaMe;
-    }
-
-    
-    public String GeT_FiRsT_NaMe() {
-        return FiRsT_NaMe;
-    }
-
-
-    public void AdD_FiNe(double fine) {
-        FiNeS_OwInG += fine;
-    }
-    
-    public double PaY_FiNe(double AmOuNt) {
-        if (AmOuNt < 0) 
-            throw new RuntimeException("Member.payFine: amount must be positive");
-        
-        double cHaNgE = 0;
-        if (AmOuNt > FiNeS_OwInG) {
-            cHaNgE = AmOuNt - FiNeS_OwInG;
-            FiNeS_OwInG = 0;
-        }
-        else 
-            FiNeS_OwInG -= AmOuNt;
-        
-        return cHaNgE;
-    }
-
-
-
-}
+package library.entities;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings("serial")
+public class Patron implements Serializable {
+
+    private long patronId;
+    private String firstName;
+    private String lastName;
+    private String emailAddress;
+    private long phoneNumber;
+    private double finesOwing;
+    private Map<Long, Loan> currentLoans;
+
+    
+    public Patron(String firstName, String lastName, String emailAddress, long phoneNumber, long memberId) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.emailAddress = emailAddress;
+        this.phoneNumber = phoneNumber;
+        this.patronId = memberId;
+        this.currentLoans = new HashMap<>();
+    }
+
+    
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("Patron:  ").append(patronId).append("\n")
+          .append("  Name:  ").append(firstName).append(" ").append(lastName).append("\n")
+          .append("  Email: ").append(emailAddress).append("\n")
+          .append("  Phone: ").append(phoneNumber)
+          .append("\n")
+          .append(String.format("  Fines Owed :  $%.2f", finesOwing))
+          .append("\n");
+        
+        for (Loan LoAn : currentLoans.values()) {
+            sb.append(LoAn).append("\n");
+        }          
+        return sb.toString();
+    }
+
+    
+    public Long getId() {
+        return patronId;
+    }
+
+    
+    public List<Loan> getLoans() {
+        return new ArrayList<Loan>(currentLoans.values());
+    }
+
+    
+    public int getNumberOfCurrentLoans() {
+        return currentLoans.size();
+    }
+
+    
+    public double finesOwed() {
+        return finesOwing;
+    }
+
+    
+    public void takeOutLoan(Loan loan) {
+        if (!currentLoans.containsKey(loan.getId())) {
+            currentLoans.put(loan.getId(), loan);
+        }
+        else {
+            throw new RuntimeException("Duplicate loan added to member");
+        }
+    }
+
+
+    public void dischargeloan(Loan loan) {
+        if (currentLoans.containsKey(loan.getId())) {
+            currentLoans.remove(loan.getId());
+        }
```

Comment:
PR#36 R1-D5-CSG#10: The argument list of any metho calls MUST NOT contain any other method calls
same hint as PR#36 R1-D4-CSG#10

Reply:
[Fixed here with this commit] (https://github.com/elliotmitchum/ITC205/pull/36/commits/3532e98039fc8d9bdc3e56579303ac073dcd90ad)